actors/metrics/src/main/java/com/ea/orbit/actors/metrics/MetricsManager.java

```diff
@@ -0,0 +1,148 @@
+        }
+    }
+
+    public <T> void registerMetric(Class clazz, String name, Supplier<T> value)
+    {
+        registry.register(MetricRegistry.name(clazz, name), (Gauge<T>) () -> value.get());
+    }
+
+    public void registerExportedMetrics(Object obj)
+    {
+        if (obj == null)
+        {
+            return;
+        }
+
+        for (Field field : obj.getClass().getDeclaredFields())
+        {
+            if (field.isAnnotationPresent(ExportMetric.class))
+            {
+                //check to see if the field is accessible.
+                if (!field.isAccessible())
+                {
+                    continue;
+                }
+                final ExportMetric annotation = field.getAnnotation(ExportMetric.class);
+                final String gaugeName = MetricRegistry.name(obj.getClass(), annotation.name());
+
+                registry.register(gaugeName, (Gauge<Object>) () -> {
+                    try
+                    {
+                        Object value = field.get(obj);
+                        return value;
+                    }
+                    catch (IllegalAccessException iae) //convert to an unchecked exception.
+                    {
+                        throw new IllegalStateException("Field " + field.getName() + " was inaccessible: " + iae.getMessage());
+                    }
+                });
+
+                logger.debug("Registered new metric for field " + field.getName() + " in class " + obj.getClass());
+            }
+        }
+
+        for (Method method : obj.getClass().getDeclaredMethods())
+        {
+            if (method.isAnnotationPresent(ExportMetric.class))
+            {
+                //methods declared as metrics must not accept parameters.
+                if (method.getParameterCount() > 0)
+                {
+                    continue;
+                }
+
+                final ExportMetric annotation = method.getAnnotation(ExportMetric.class);
+                final String gaugeName = MetricRegistry.name(obj.getClass(), annotation.name());
+
+                registry.register(gaugeName, (Gauge<Object>) () -> {
+
+                    try
+                    {
+                        Object value = method.invoke(obj, new Object[0]);
+                        return value;
+                    }
+                    catch (IllegalAccessException iae) //convert to an unchecked exception.
+                    {
+                        throw new IllegalStateException("Method " + method.getName() + " was inaccessible: " + iae.getMessage());
+                    }
+                    catch (InvocationTargetException ite)
+                    {
+                        throw new UncheckedException("Invocation of method " + method.getName() + " failed", ite.getTargetException());
+                    }
+                });
+
+                logger.debug("Registered new metric for method " + method.getName() + " in class " + obj.getClass());
```

Comment:
Not really any performance benefit here but I generally recommend to wrap debug calls with isdebugenabled.